### ISSUE TITLE
Generics produce invalid OpenAPI identifiers

### ISSUE THREAD

**Issue by Eugeny** [poem-web/poem#318]
Having a generic type with `#[derive(Object)]` will produce identifiers like `PaginatedResponse<SessionSnapshot>` which look nice, but are not valid according to the spec:

> Component names can only contain the characters A-Z a-z 0-9 - . _

It would be nicer to allow renaming them at the specialization site with an `[oai(rename)]`, or allow forcing the definition to be inlined in the spec.

**Comment by sunli829** [comment]
It works fine in `Swagger UI`/`Redoc`/`Rapidoc`, maybe some code generation tools don't recognize it. 🙂

**Comment by itsbalamurali** [comment]
@sunli829 what's the best workaround? Currently, we use [openapi-generator](http://openapi-generator.tech). Getting the same issue as @Eugeny did.

**Comment by Eugeny** [comment]
Maybe the Object derive macro could also accept an `#[oai(generic_name_format="{}<{}>")]` and use to generate the name

**Comment by itsbalamurali** [comment]
@Eugeny or @sunli829 any plans on landing this sooner?

**Comment by sunli829** [comment]
Currently the only solution is to avoid using generics. 🙂

**Comment by itsbalamurali** [comment]
@Eugeny are you interested in implementing this?

**Comment by Eugeny** [comment]
No, I've already worked around it by just modifying the generated JSON schema in my code.

**Comment by itsbalamurali** [comment]
Just pinging to keep the issue alive.

**Comment by itsbalamurali** [comment]
@sunli829 Do you have some time to look at this?

**Comment by sunli829** [comment]
This problem can currently only be solved by avoiding the use of generics. 🙂

**Comment by itsbalamurali** [comment]
pinging to keep the issue alive.

**Comment by pandranki-skytv** [comment]
ping again :)
